Casks/j/java@beta.rb

```diff
@@ -1,20 +1,24 @@
 cask "java@beta" do
   arch arm: "aarch64", intel: "x64"
 
-  version "24,35"
-  sha256 arm:   "04436b666122e650a0bbd31941b28a068f4fee2c84bb0f839eefe0ca19c67308",
-         intel: "44792f55d0e05ad8cea40bd85363a82615de6d8d24427a2371681ccfef79668a"
+  version "25,9"
+  sha256 arm:   "6a55f389999988813c03e8928685f6fe8996e56ee85536869dd9d1ba7dc634f9",
+         intel: "b8df0f3869fa86f4c1c4329f6d717982b547c404d0713f7e1fa6ffbfc8afcfec"
 
   url "https://download.java.net/java/early_access/jdk#{version.major}/#{version.csv.second}/GPL/openjdk-#{version.csv.first}-ea+#{version.csv.second}_macos-#{arch}_bin.tar.gz"
   name "OpenJDK Early Access Java Development Kit"
   desc "Early access development kit for the Java programming language"
   homepage "https://jdk.java.net/"
 
   livecheck do
-    url "https://jdk.java.net/#{version.major}/"
+    url :homepage
     regex(%r{href=.*?/GPL/openjdk-(\d+)-ea\+(\d+)_macos-#{arch}_bin\.t}i)
     strategy :page_match do |page, regex|
-      page.scan(regex).map { |match| "#{match[0]},#{match[1]}" }
+      latest_major = page[/Early\s*access:.*?JDK\s*(\d+)/i, 1]
+      next if latest_major.blank?
+
+      version_page = Homebrew::Livecheck::Strategy.page_content("https://jdk.java.net/#{latest_major}/")
```

Comment:
```suggestion
      ea_text = page[%r{<h\d+>\s*Early\s+access:.+?</h\d+>}im]
      next unless ea_text

      highest_version = ea_text.scan(/JDK\s*(\d+)/i)
                               .flatten
                               .uniq
                               .map { |v| Version.new(v) }
                               .max
      next unless highest_version

      version_page = Homebrew::Livecheck::Strategy.page_content("https://jdk.java.net/#{highest_version}/")
```

Hmm, this only works as expected if the early access versions remain in descending order (i.e., highest to lowest). I try to avoid relying on things that can change like this, so it's better if we extract all the versions from the early access text and find the highest ourself (we do something similar in a few other formulae/casks that make another network request in the `strategy` block).

This complicates the `strategy` block logic a bit (and it can fail if upstream changes the related HTML structure) but I think it's preferable over relying on a specific formatting convention.